pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/v2/FloatMutableDictionary.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.realtime.impl.dictionary.v2;
+
+import java.util.Arrays;
+import javax.annotation.Nonnull;
+
+
+public class FloatMutableDictionary extends BaseMutableDictionary {
+  private float _min = Float.MAX_VALUE;
+  private float _max = Float.MIN_VALUE;
+
+  @Override
+  public int indexOf(Object rawValue) {
+    if (rawValue instanceof String) {
+      return getDictId(Float.valueOf((String) rawValue));
+    } else {
+      return getDictId(rawValue);
+    }
+  }
+
+  @Override
+  public void index(@Nonnull Object rawValue) {
+    if (rawValue instanceof Float) {
```

Comment:
This check is for multi-value columns, in which case the rawValue is an Object[]